### PR TITLE
feat(remote): custom certificate validation with single execution path - fixes mTLS asymmetry bug

### PR DESCRIPTION
**Pull request by Aaronontheweb**
## Summary

This PR implements a comprehensive programmatic certificate validation API for Akka.NET remote TLS transport, addressing the critical asymmetric hostname validation bug and providing high-level configuration flexibility.

### Key Changes

**6 commits total:**

1. **feat(remote)**: `CertificateValidationCallback` delegate and `CertificateValidation` helper factory
   - New `CertificateValidationCallback` delegate for user-facing validation logic
   - 7 helper factory methods: `ValidateChain`, `ValidateHostname`, `PinnedCertificate`, `ValidateSubject`, `ValidateIssuer`, `Combine`, `ChainPlusThen`
   - Adapter bridge between user delegates and DotNetty's low-level validation

2. **fix(remote)**: Single execution path for validation with hostname validation asymmetry fix
   - Fixes critical bug: client-enforced hostname validation was ignored by server
   - Server now enforces hostname validation when client requires it
   - Eliminates dual branching logic via single `CustomValidator ?? ComposeValidatorFromSettings()` path

3. **fix(remote)**: Reject missing client certificates in server-side mutual TLS validation
   - Explicit null certificate rejection when mTLS required
   - Test `Mutual_TLS_should_fail_when_client_has_no_certificate` now passes

4. **docs**: Programmatic certificate validation examples and consolidated security documentation
   - Added 5 code examples to `TlsConfigurationSample.cs` with #region tags
   - Created "Validation Strategies: HOCON vs Programmatic" section with decision matrix
   - Unified "Hostname Validation" and "Mutual TLS" sections to reduce duplication

5. **fix**: Corrected compilation errors in documentation examples
   - Added `Akka.Remote` reference to `Akka.Docs.Tests` project
   - Wrapped examples with `#if NET6_0_OR_GREATER` for framework compatibility
   - Fixed API usage: params vs arrays, correct delegate signatures

6. **fix**: Wire CustomValidator through SslSettings and add comprehensive integration tests ⭐ **CRITICAL**
   - Fixed `DotNettySslSetup.Settings` to pass `CustomValidator` to `SslSettings` constructor
   - Removed unused `ValidateCertificateHostnameMatch` method (489-542 lines)
   - Added 3 integration tests that actually validate CustomValidator functionality works end-to-end

### Test Results
- All 327 unit tests passing ✓ (324 existing + 3 new)
- 5 tests skipped (expected)
- All 42 DotNetty tests pass ✓
- DocFX metadata build: 0 warnings/errors ✓
- DocFX full build: 0 warnings/errors ✓

### New Integration Tests (Commit 6)
1. **`CustomValidator_that_accepts_should_allow_connection`**
   - Verifies CustomValidator callback is invoked during TLS handshake
   - Verifies acceptance allows successful connection

2. **`CustomValidator_that_rejects_should_prevent_connection`**
   - Verifies CustomValidator rejection prevents connection
   - Verifies callback was invoked even when rejecting

3. **`DotNettySslSetup_should_pass_CustomValidator_to_SslSettings`**
   - Unit test verifying CustomValidator is properly wired through

### Files Modified
- `DotNettyTransport.cs` - Single execution path + null certificate rejection + removed unused method
- `DotNettyTransportSettings.cs` - CertificateValidationCallback + CertificateValidation factory
- `DotNettySslSetup.cs` - CustomValidator property + wiring to SslSettings
- `SslSettings.cs` - CustomValidator property
- `DotNettySslSetupSpec.cs` - 3 new integration tests for CustomValidator
- `TlsConfigurationSample.cs` - 5 programmatic examples (net6.0+)
- `Akka.Docs.Tests.csproj` - Akka.Remote reference
- `security.md` - Consolidated validation strategies + decision matrix

### Documentation
The new API provides:
- High-level programmatic control over certificate validation
- 5 usage patterns: chain validation, hostname validation, certificate pinning, subject validation, issuer validation
- Composable validators via `Combine` and `ChainPlusThen`
- Custom validation logic with `ChainPlusThen` for complex scenarios

### Backwards Compatibility
✓ Fully backwards compatible - all changes are additive
✓ Existing HOCON-based configuration unaffected
✓ New API is optional; users can continue using HOCON

## Test Plan

- [x] All unit tests pass (327/327)
- [x] Multi-node tests pass
- [x] Mutual TLS validation fixed and tested
- [x] CustomValidator wiring fixed and tested
- [x] CustomValidator acceptance/rejection tested end-to-end
- [x] Documentation builds without warnings/errors
- [x] Code examples compile correctly
- [x] API compatibility maintained

### CODE REVIEW

File: src/core/Akka.Remote/Transport/DotNetty/DotNettySslSetup.cs [hunk 1]
```diff
@@ -77,5 +103,13 @@ public DotNettySslSetup(X509Certificate2 certificate, bool suppressValidation, b
     /// </summary>
     public bool ValidateCertificateHostname { get; }
 
+    /// <summary>
+    /// Custom certificate validation callback for advanced validation scenarios.
+    /// When provided, this callback takes precedence over config-based validation.
+    /// Use with CertificateValidation helper factory to combine multiple validation strategies.
+    /// Example: CertificateValidation.Combine(ValidateChain(log), PinnedCertificate(thumbprints))
+    /// </summary>
+    public CertificateValidationCallback? CustomValidator { get; }
```
Comment (Arkatufus) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettySslSetup.cs:113]: Where is this being used? It is not being passed into the SslSettings which is the thing that is actually being used by DotNettyTransport.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs [hunk 2]
```diff
@@ -464,6 +457,90 @@ private void SetServerPipeline(IChannel channel)
...
+                return dotnettyCallback(null, cert, chain, errors);
+            };
+        }
+
+        /// <summary>
+        /// Validates that a certificate's CN or Subject Alternative Name matches the expected hostname.
+        /// Supports wildcard certificates and IP addresses.
+        /// Uses reflection for compatibility across .NET Framework, .NET Standard 2.0, and .NET 6+
+        /// </summary>
+        private bool ValidateCertificateHostnameMatch(X509Certificate2 certificate, string expectedHostname)
```
Comment (Arkatufus) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs]: Is this being used anywhere?

**Review by Arkatufus — CHANGES_REQUESTED** [review]
Some questions

**Comment by Aaronontheweb** [comment]
## Review Comments Addressed

Thank you for catching these critical issues! I've pushed fixes in commit 29ec21ce2.

### Issue 1: CustomValidator not passed to SslSettings (Line 112)
**Fixed**: Line 114 in `DotNettySslSetup.cs` now passes `CustomValidator` to the SslSettings constructor:
```csharp
internal SslSettings Settings => new SslSettings(Certificate, SuppressValidation, RequireMutualAuthentication, ValidateCertificateHostname, CustomValidator);
```

### Issue 2: ValidateCertificateHostnameMatch unused (Line 494)
**Fixed**: Removed the unused method (lines 489-542 in DotNettyTransport.cs). Hostname validation is already handled by `TlsValidationCallbacks.Create` which filters `RemoteCertificateNameMismatch` errors based on configuration.

### Issue 3: Tests don't actually test CustomValidator functionality  
**Fixed**: Added 3 comprehensive integration tests to `DotNettySslSetupSpec.cs`:

1. **`CustomValidator_that_accepts_should_allow_connection`**
   - Creates ActorSystem with CustomValidator that accepts all certificates
   - Verifies callback is invoked during TLS handshake
   - Verifies connection succeeds

2. **`CustomValidator_that_rejects_should_prevent_connection`**
   - Creates ActorSystem with CustomValidator that rejects all certificates
   - Verifies callback is invoked  
   - Verifies connection fails (no message arrives)

3. **`DotNettySslSetup_should_pass_CustomValidator_to_SslSettings`**
   - Unit test verifying CustomValidator is properly wired through

**Test Results**: All 3 new tests pass ✓

These tests actually validate that:
- CustomValidator callbacks are invoked during TLS handshake
- CustomValidator acceptance/rejection affects connection behavior
- The wiring from DotNettySslSetup → SslSettings → DotNettyTransport is correct

**Comment by Aaronontheweb** [comment]
## Concern: Mixing HOCON with DotNettySslSetup

You're right to be concerned about this. The current precedence rules are:

**DotNettySslSetup takes precedence over HOCON** (fixed in PR #7918)

However, this could be confusing for users. Here are a few options to clarify:

### Option 1: Add Warning Logs
Add a warning when both are configured:
```csharp
if (setup.HasValue && hoconConfig.HasCertificate) 
{
    log.Warning("Both DotNettySslSetup and HOCON SSL config detected. DotNettySslSetup takes precedence.");
}
```

### Option 2: Documentation Clarity
Already addressed in this PR - the security.md includes a decision matrix showing "Validation Strategies: HOCON vs Programmatic" with clear guidance on when to use each.

### Option 3: Make it Mutually Exclusive
Throw an exception if both are configured:
```csharp
if (setup.HasValue && hoconConfig.HasCertificate)
{
    throw new ConfigurationException(
        "Cannot configure both DotNettySslSetup and HOCON SSL settings. Choose one approach.");
}
```

**Recommendation**: I'd suggest **Option 2 (current approach)** with maybe **Option 1 (warning logs)** added as a safety net. Making it mutually exclusive (Option 3) would be a breaking change and might be too strict.

The precedence behavior (setup > HOCON) makes sense because:
1. Programmatic config is more explicit/intentional
2. Allows per-ActorSystem customization even with shared HOCON config files
3. Matches the pattern of other Setup classes in Akka.NET

What's your preference? I can add warning logs if that would help.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettySslSetup.cs (hunk 1 above)
Comment (Aaronontheweb) [reply src/core/Akka.Remote/Transport/DotNetty/DotNettySslSetup.cs:113]: Resolved

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs (hunk 2 above)
Comment (Aaronontheweb) [reply src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs]: Nope, removed it.

File: docs/articles/remoting/security.md
Comment (Aaronontheweb) [inline docs/articles/remoting/security.md:1]: Probably need to simplify and clean this up in a separate PR, but the new cert validation strategies are documented here.

File: src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveRemote.DotNet.verified.txt [hunk 3]
```diff
@@ -860,12 +860,29 @@ namespace Akka.Remote.Transport
 }
 namespace Akka.Remote.Transport.DotNetty
 {
+    [System.Runtime.CompilerServices.NullableAttribute(0)]
+    public class static CertificateValidation
```
Comment (Aaronontheweb) [inline src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveRemote.DotNet.verified.txt:864]: New set of helpers for defining certificate validation callbacks

File: src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveRemote.DotNet.verified.txt [hunk 4]
```diff
@@ -860,12 +860,29 @@ namespace Akka.Remote.Transport
...
+    }
+    public delegate bool CertificateValidationCallback(System.Security.Cryptography.X509Certificates.X509Certificate2 certificate, System.Security.Cryptography.X509Certificates.X509Chain chain, string remotePeer, System.Net.Security.SslPolicyErrors errors, Akka.Event.ILoggingAdapter log);
     public sealed class DotNettySslSetup : Akka.Actor.Setup.Setup
     {
         public DotNettySslSetup(System.Security.Cryptography.X509Certificates.X509Certificate2 certificate, bool suppressValidation) { }
         public DotNettySslSetup(System.Security.Cryptography.X509Certificates.X509Certificate2 certificate, bool suppressValidation, bool requireMutualAuthentication) { }
         public DotNettySslSetup(System.Security.Cryptography.X509Certificates.X509Certificate2 certificate, bool suppressValidation, bool requireMutualAuthentication, bool validateCertificateHostname) { }
+        public DotNettySslSetup(System.Security.Cryptography.X509Certificates.X509Certificate2 certificate, bool suppressValidation, bool requireMutualAuthentication, [System.Runtime.CompilerServices.NullableAttribute(2)] Akka.Remote.Transport.DotNetty.CertificateValidationCallback customValidator) { }
```
Comment (Aaronontheweb) [inline src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveRemote.DotNet.verified.txt:886]: CTOR overload for adding the custom `CertificationValidationCallabck` to the `DotNettySslSetup`. This will need to be integrated into Akka.Hosting.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs [hunk 5]
```diff
@@ -588,6 +649,213 @@ public static RemoteCertificateValidationCallback AcceptAll()
...
+    /// Helpers return delegates that can be composed or used standalone.
+    /// Each helper creates a CertificateValidationCallback that can be passed to DotNettySslSetup.
+    /// </summary>
+    public static class CertificateValidation
+    {
+        /// <summary>
+        /// Validate certificate chain against system CA store.
+        /// Use for: CA-signed certificates in production.
+        /// </summary>
+        public static CertificateValidationCallback ValidateChain(
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs:540]: XML-DOC comment is accurate, but yes: validates the CA chain.

**Review by Aaronontheweb — COMMENTED** [review]
First pass at a review - found a bunch of stuff to fix already.

File: src/core/Akka.Docs.Tests/Configuration/TlsConfigurationSample.cs [hunk 6]
```diff
@@ -80,5 +83,133 @@ public class TlsConfigurationSample
...
+        /// Example of programmatic mutual TLS setup using DotNettySslSetup with custom validation.
+        /// This allows full programmatic control over certificate validation logic.
+        /// </summary>
+        public static void ProgrammaticMutualTlsSetup()
+        {
+            // Load or obtain your certificate
+            var certificate = new X509Certificate2("path/to/certificate.pfx", "password");
+
+            // Create custom validator combining multiple validation strategies
+            var customValidator = CertificateValidation.Combine(
```
Comment (Aaronontheweb) [inline src/core/Akka.Docs.Tests/Configuration/TlsConfigurationSample.cs:98]: if you want stricter TLS validation, this is a good example of how to do it with certificate pinning.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettySslSetup.cs [hunk 7]
```diff
@@ -5,6 +5,7 @@
 // </copyright>
 //-----------------------------------------------------------------------
 
+#nullable enable
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettySslSetup.cs:8]: Enabled nullability here in order to catch errors in certification validation / config.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs [hunk 8]
```diff
@@ -5,6 +5,7 @@
 // </copyright>
 //-----------------------------------------------------------------------
 
+#nullable enable
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs:8]: Nullability enabled to catch SSL certificate errors, but caught some other potential nasties too.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs [hunk 9]
```diff
@@ -357,22 +358,25 @@ private void SetClientPipeline(IChannel channel, Address remoteAddress)
...
-                // Build validation callback using type-safe factory methods
-                // These settings are independent and can be combined:
-                // - suppressValidation: Controls chain/CA validation (for self-signed certs)
-                // - validateCertificateHostname: Controls hostname matching (for per-node certs, IPs, etc.)
-                var chainValidation = Settings.Ssl.SuppressValidation
-                    ? ChainValidationMode.IgnoreChainErrors
-                    : ChainValidationMode.ValidateChain;
+                // Compose validator: either use custom validator or build from config settings
+                // This ensures a single execution path through validation logic
+                var validator = Settings.Ssl.CustomValidator ?? ComposeValidatorFromSettings();
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs:363]: Use either the user-supplied certificate validation code or build one based on the `SslSettings`.

File: src/core/Akka.Remote.Tests/Transport/CertificateValidationHelpersSpec.cs [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+
+            // Act
+            var result = validator(cert, null, "test-peer", SslPolicyErrors.None, _log);
+
+            // Assert
+            Assert.True(result); // Should accept because valid thumbprint is in the list
+        }
+
+        [Fact(DisplayName = "PinnedCertificate should be case-insensitive for thumbprints")]
+        public void PinnedCertificate_should_be_case_insensitive()
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote.Tests/Transport/CertificateValidationHelpersSpec.cs:89]: See the comment on the actual code being tested here, but there's no security issue due to thumbprints being HEX, so case sensitivity doesn't impact their underlying values

File: src/core/Akka.Remote.Tests/Transport/CertificateValidationHelpersSpec.cs [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+        [Fact(DisplayName = "Combine should handle empty validators array")]
+        public void Combine_should_handle_empty_validators()
+        {
+            // Act & Assert - Should throw ArgumentException
+            Assert.Throws<ArgumentException>(() => CertificateValidation.Combine());
+            Assert.Throws<ArgumentException>(() => CertificateValidation.Combine(new CertificateValidationCallback[0]));
+        }
+
+        [Fact(DisplayName = "Combine should short-circuit on first failure")]
+        public void Combine_should_short_circuit_on_first_failure()
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote.Tests/Transport/CertificateValidationHelpersSpec.cs:215]: Keep it simple and short-circuit. It's what ASP.NET Core et al do.

File: src/core/Akka.Remote.Tests/Transport/DotNettySslSetupSpec.cs [hunk 12]
```diff
@@ -247,6 +242,526 @@ public void DotNettySslSetup_should_override_HOCON_certificate()
...
 
+        [Fact(DisplayName = "DotNettySslSetup with CustomValidator that accepts should allow connection")]
+        public async Task CustomValidator_that_accepts_should_allow_connection()
+        {
+            var validatorCalled = false;
+
+            var certificate = new X509Certificate2(ValidCertPath, Password, X509KeyStorageFlags.DefaultKeySet);
+
+            // Custom validator that accepts all certificates
+            CertificateValidationCallback customValidator = (cert, chain, peer, errors, log) =>
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote.Tests/Transport/DotNettySslSetupSpec.cs:253]: This is like a hand-rolled equivalent of "suppress validation"

File: src/core/Akka.Remote.Tests/Transport/DotNettySslSetupSpec.cs [hunk 13]
```diff
@@ -247,6 +242,526 @@ public void DotNettySslSetup_should_override_HOCON_certificate()
...
+            }, TimeSpan.FromSeconds(30), TimeSpan.FromMilliseconds(100));
+        }
+
+        [Fact(DisplayName = "CertificateValidation.PinnedCertificate should reject certificates with non-matching thumbprint")]
+        public async Task PinnedCertificate_should_reject_non_matching_thumbprint()
+        {
+            var certificate = new X509Certificate2(ValidCertPath, Password, X509KeyStorageFlags.DefaultKeySet);
+
+            // Create validator that pins to a DIFFERENT thumbprint (connection should fail)
+            var validator = CertificateValidation.PinnedCertificate("0000000000000000000000000000000000000000");
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote.Tests/Transport/DotNettySslSetupSpec.cs:461]: Non-existent thumbprint

File: src/core/Akka.Docs.Tests/Akka.Docs.Tests.csproj [hunk 14]
```diff
@@ -10,6 +10,7 @@
     <ProjectReference Include="..\Akka.Coordination.Tests\Akka.Coordination.Tests.csproj" />
     <ProjectReference Include="..\Akka\Akka.csproj" />
     <ProjectReference Include="..\Akka.Persistence\Akka.Persistence.csproj" />
+    <ProjectReference Include="..\Akka.Remote\Akka.Remote.csproj" />
```
Comment (Aaronontheweb) [inline src/core/Akka.Docs.Tests/Akka.Docs.Tests.csproj]: this is not strictly necessary

File: src/core/Akka.Remote.Tests/Transport/CertificateValidationHelpersSpec.cs
Comment (Aaronontheweb) [inline src/core/Akka.Remote.Tests/Transport/CertificateValidationHelpersSpec.cs:1]: Handles a range of certificate validation edge cases to ensure that they're all caught.

**Review by Aaronontheweb — COMMENTED** [review]
Have another pass yet to go - hardening the mTLS end to end specs now.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs [hunk 15]
```diff
@@ -357,22 +358,25 @@ private void SetClientPipeline(IChannel channel, Address remoteAddress)
...
+                {
+                    // Convert X509Certificate to X509Certificate2 if needed
+                    var x509Cert = cert as X509Certificate2 ?? (cert != null ? new X509Certificate2(cert) : null);
+                    return validator(x509Cert, chain, remoteAddress.ToString(), errors, Log);
+                };
 
                 if (Settings.Ssl.RequireMutualAuthentication)
                 {
+                    // Mutual TLS requires a certificate to be configured
+                    if (certificate == null)
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs:377]: This is a startup error / misconfiguration.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs [hunk 16]
```diff
@@ -464,6 +462,29 @@ private void SetServerPipeline(IChannel channel)
...
+        private CertificateValidationCallback ComposeValidatorFromSettings()
+        {
+            // Build validator from configuration settings
+            // Note: SuppressValidation and ValidateCertificateHostname are independent settings
+            var suppressChain = Settings.Ssl.SuppressValidation;
+            var validateHostname = Settings.Ssl.ValidateCertificateHostname;
+
+            return suppressChain switch
+            {
+                true when validateHostname => CertificateValidation.ValidateHostname(log: Log),
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettyTransport.cs:480]: if we are suppressing the CA chain but are still validating on hostname, we will do that here.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs [hunk 17]
```diff
@@ -143,9 +145,26 @@ public static DotNettyTransportSettings Create(ActorSystem system)
...
+
+            // Warn if both DotNettySslSetup and HOCON SSL are configured (DotNettySslSetup takes precedence)
+            if (sslSettings != null && config.GetBoolean("enable-ssl"))
+            {
+                var sslConfig = config.GetConfig("ssl");
+                // Only warn if HOCON has explicit certificate configuration
+                var hasCertPath = sslConfig.HasPath("certificate.path") && !string.IsNullOrWhiteSpace(sslConfig.GetString("certificate.path"));
+                var hasCertThumbprint = sslConfig.HasPath("certificate.thumbprint") && !string.IsNullOrWhiteSpace(sslConfig.GetString("certificate.thumbprint"));
+
+                if (hasCertPath || hasCertThumbprint)
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs:160]: Issue a warning if a user duped their configuration between HOCON and the `DotNettySslSetup`.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs [hunk 18]
```diff
@@ -459,133 +501,297 @@ private SslSettings(string certificatePath, string certificatePassword, X509KeyS
...
         /// </summary>
-        /// <param name="chainValidation">Controls certificate chain/CA validation</param>
-        /// <param name="hostnameValidation">Controls hostname matching validation</param>
-        /// <param name="log">Logger for validation failures</param>
-        /// <returns>Validation callback configured according to parameters</returns>
-        public static RemoteCertificateValidationCallback Create(
-            ChainValidationMode chainValidation,
-            HostnameValidationMode hostnameValidation,
-            ILoggingAdapter log)
+        public static CertificateValidationCallback PinnedCertificate(
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs:598]: Only accept certs with a given set of thumb prints.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs [hunk 19]
```diff
@@ -459,133 +501,297 @@ private SslSettings(string certificatePath, string certificatePassword, X509KeyS
...
+            {
+                if (cert == null)
+                {
+                    (log ?? nonClosureLog).Error(
+                        "Hostname validation failed for {0}: certificate is null",
+                        peer);
+                    return false;
+                }
+
+                var hostname = expectedHostname ?? peer;
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs:581]: fall back to validating our outbound peer's connect hostname if the user didn't provide one.

Comment (Aaronontheweb) [reply src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs:581]: This is consistent with earlier behavior.

**Review by Aaronontheweb — COMMENTED** [review]
Done

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs
Comment (Arkatufus) [inline src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs]: Not related to this PR but this is missed on the previous PR. SslPolicyErrors.None is 0, anything "bitwise and" 0 always returns a 0, so this if statement always returns a false.

Comment (Aaronontheweb) [reply src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs]: That's not a logical boolean though, it's a bit-wise and `&`

**Review by Arkatufus — APPROVED** [review]
Changes looks good to me, but I noticed something that we missed from the previous PR.

File: src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs
Comment (Aaronontheweb) [reply src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs]: Ah misread your comment - let me go double check that

Comment (Arkatufus) [reply src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs]: yes, I meant bitwise and

Comment (Aaronontheweb) [reply src/core/Akka.Remote/Transport/DotNetty/DotNettyTransportSettings.cs]: Ok, this has been resolved
